src/vhdlConvertor/typeDeclarationParser.cpp

```diff
@@ -0,0 +1,238 @@
+#include <hdlConvertor/vhdlConvertor/typeDeclarationParser.h>
+
+namespace hdlConvertor {
+namespace vhdl {
+
+using namespace hdlConvertor::hdlObjects;
+using vhdlParser = vhdl_antlr::vhdlParser;
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitType_declaration(
+		vhdlParser::Type_declarationContext *ctx) {
+	//type_declaration:
+	//      full_type_declaration
+	//      | incomplete_type_declaration
+	//;
+	auto ftd = ctx->full_type_declaration();
+	if (ftd) 
+		return visitFull_type_declaration(ftd);
+	else
+		return visitIncomplete_type_declaration(ctx->incomplete_type_declaration());
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitIncomplete_type_declaration(
+		vhdlParser::Incomplete_type_declarationContext *ctx) {
+	//incomplete_type_declaration: KW_TYPE identifier SEMI;
+	NotImplementedLogger::print(
+			"TypeDeclarationParser.visitIncomplete_type_declaration", ctx);
+	return nullptr;
+	//todo: create an hdltc_error type
+}
+
+static std::unique_ptr<hdlObjects::HdlTypeDec> VhdlTypeDeclarationParser::visitFull_type_declaration(
+		vhdlParser::Full_type_declarationContext *ctx) {
+	//full_type_declaration:
+	//      KW_TYPE identifier KW_IS type_definition SEMI
+	//;
+	return visitType_definition(
+		VhdlLiteralParser::getIdentifierStr(ctx->identifier()), 
```

Comment:
I'm still wrapping my head around modern C++.  `getIdentifierStr` returns a `std::string` by value.  I'm hoping that the rvalue reference argument in the sub-parsing methods will "forward" this value to the appropriate constructor when we finally figure out which data type subclass we want.